ark/type/keywords/string/base64.ts

```diff
@@ -0,0 +1,34 @@
+import type { Module, Submodule } from "../../module.ts"
+import type { Predicate, of } from "../inference.ts"
+import { arkModule } from "../utils.ts"
+import { regexStringNode } from "./utils.ts"
+
+export const base64 = arkModule({
+	root: regexStringNode(
+		/^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/,
+		"base64-encoded"
+	),
+	url: regexStringNode(
+		/^(?:[A-Za-z0-9_-]{4})*(?:[A-Za-z0-9_-]{2}(?:==|%3D%3D)?|[A-Za-z0-9_-]{3}(?:=|%3D)?)?$/,
+		"base64-encoded"
```

Comment:
Sorry if I was misleading, this should still be `"a base64-encoded url"` to distinguish it from above

Reply:
Perhaps something like `"a URL-friendly base64-encoded string"`? A base64-encoded url sounds like it would be something else.

Reply:
Yeah you're right, I guess based on the language from the spec it should be "URL-safe."

Reply:
I think `"base64url-encoded"` is the way to go as that's what the [RFC specification](https://datatracker.ietf.org/doc/html/rfc4648#section-5) calls the variant.

Reply:
Yeah was reading it from your original description. That works for me